src/components/user/user.service.ts

```diff
@@ -402,15 +402,17 @@ export class UserService {
     // Update roles
     if (input.roles) {
       const newRoles = difference(input.roles, user.roles.value);
-      await this.db
-        .query()
-        .match([
-          node('user', ['User', 'BaseNode'], {
-            id: input.id,
-          }),
-        ])
-        .create([...this.roleProperties(newRoles)])
-        .run();
+      if (newRoles.length) {
+        await this.db
+          .query()
+          .match([
+            node('user', ['User', 'BaseNode'], {
+              id: input.id,
+            }),
+          ])
+          .create([...this.roleProperties(newRoles)])
+          .run();
+      }
```

Comment:
What about removing existing roles not present in the input?

Reply:
Okay, I will remove old roles.